InterfaceFortran/TasmanianSG.f90

```diff
@@ -154,501 +169,732 @@ Module TasmanianSG
                         tsg_acc_none     = 0, tsg_acc_cpu_blas  = 1, tsg_acc_gpu_cublas = 2, &
                         tsg_acc_gpu_cuda = 3, tsg_acc_gpu_magma = 4
 PRIVATE
-  INTEGER :: rows, cols, length
-  DOUBLE PRECISION, pointer :: matrix(:,:), vector(:)
-  CHARACTER, pointer :: string(:)
-CONTAINS
-!=======================================================================
-SUBROUTINE tsgInitialize()
-  CALL tsgbeg()
-END SUBROUTINE tsgInitialize
-!=======================================================================
-SUBROUTINE tsgFinalize()
-  CALL tsgend()
-END SUBROUTINE tsgFinalize
-!=======================================================================
-FUNCTION tsgNewGridID() result(newid)
-  INTEGER :: newid
-  CALL tsgnew(newid)
-END FUNCTION tsgNewGridID
-!=======================================================================
-SUBROUTINE tsgFreeGridID(gridID)
-  INTEGER :: gridID
-  CALL tsgfre(gridID)
-END SUBROUTINE tsgFreeGridID
+  integer :: rows, cols, length
+  double precision, pointer :: matrix(:,:), vector(:)
+  character, pointer :: string(:)
+contains
+!=======================================================================
+function tsgIsGlobal(id) result(res)
+  integer :: id, flag
+  logical :: res
+  call tsgisg(id, flag)
+  if (flag .eq. 1) then
```

Comment:
* `tsgisg()` is a C function returning `flag` which is in fact an integer, basically an integer is used for a boolean/logical
* in C programming standard, integer `0` is considered `false` while `not-0` is considered `true`
* when using an integer as a C boolean/logical, you should never have statement such as `(flag .eq. 1)` as flag of 2, 3, and so on, is also technically `true`
* you wrote the `tsgisg()` and you know what it returns, but conform to the C standard, i.e., use
```
if (flag .ne. 0)
   res = .true.
...
```
or
```
if (flag .eq. 0)
  res = .false.
...
```

Reply:
* fix the rest of the functions too

Reply:
* Yes, I want to be compliant with the C interface. I know this is not a conflict with your code, but it can confuse someone else. We should always comply with the standards of various languages, whenever we can and whenever we know what the actual standard is. We know this one standard.
* In most cases you can just change `.eq.` with `.ne.` or `==` to `!=`